pdns/recursordist/mtasker.hh

```diff
@@ -332,18 +334,18 @@ void MTasker<Key, Val, Cmp>::yield()
 template <class EventKey, class EventVal, class Cmp>
 int MTasker<EventKey, EventVal, Cmp>::sendEvent(const EventKey& key, const EventVal* val)
 {
-  typename waiters_t::iterator waiter = d_waiters.find(key);
+  auto waiter = d_waiters.find(key);
 
   if (waiter == d_waiters.end()) {
-    // cerr<<"Event sent nobody was waiting for! " <<key << endl;
     return 0;
   }
   d_waitstatus = Answer;
   if (val) {
     d_waitval = *val;
   }
   d_tid = waiter->tid; // set tid
-  d_eventkey = waiter->key; // pass waitEvent the exact key it was woken for
+  d_eventkey = std::move(waiter->key); // pass waitEvent the exact key it was woken for
+  d_used = false;
   auto userspace = std::move(waiter->context);
   d_waiters.erase(waiter); // removes the waitpoint
```

Comment:
Calling `erase()` after the key has been moved worries me. In theory it should be OK given that we use an iterator, but it feels brittle. Do we really need to move the key here, given that the entry is going to be erased anyway?

Reply:
On second thought, I agree.